bauble/src/types.rs

```diff
@@ -146,6 +149,22 @@ impl Variant {
         self.extra_validation = Some(validation);
         self
     }
+
+    pub fn with_default(mut self, value: UnspannedVal) -> Self {
+        self.default = Some(value);
+        self
+    }
+}
+
+#[derive(Debug)]
+pub enum TypeSystemError<'a> {
+    ToBeAssigned(Vec<(TypeId, TypePath<&'a str>)>),
+    NotInstantiable {
+        ty_id: TypeId,
+        ty: TypePath<&'a str>,
+    },
+    InstantiableErrors,
+    ConstructInequality(UnspannedVal, UnspannedVal),
 }
```

Comment:
It would be nice for UX if this type used `thiserror` to provide friendly error messages. Could also manually implement `Display` if you want to avoid adding proc macro dependencies that might increase compile times.

Reply:
Implemented display for it, felt a bit unnecessary to pull in `thiserror` for just this small enum.